File: app/src/main/java/com/crimson/addect/dagger/component/ActivityComponent.java
package com.crimson.addect.dagger.component;

import com.crimson.addect.dagger.module.ActivityModule;
import com.crimson.addect.dagger.scope.ActivityScope;
import com.crimson.addect.feature.playscreen.PlayActivity;
import com.crimson.addect.feature.startscreen.StartScreenActivity;
import dagger.Component;
import org.jetbrains.annotations.NotNull;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = { ActivityModule.class })
public interface ActivityComponent {

  void inject(@NotNull StartScreenActivity startScreenActivity);

  void inject(@NotNull PlayActivity playActivity);
}

File: app/src/main/java/com/crimson/addect/dagger/component/ActivityComponent.kt
package com.crimson.addect.dagger.component

import com.crimson.addect.dagger.module.ActivityModule
import com.crimson.addect.dagger.scope.ActivityScope
import com.crimson.addect.feature.playscreen.PlayActivity
import com.crimson.addect.feature.startscreen.StartScreenActivity
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {

  fun inject(startScreenActivity: StartScreenActivity)

  fun inject(playActivity: PlayActivity)
}
